attempt for #451

SoumyajitKarmakar opened the pull request:
Modified the env.reset() function to add the reward_space and observation_space parameters. Modified the reset() func of the subclasses. Added a OptionalArgumentValue class in gym_type_hints.py for the default value.

Fixes #451

SoumyajitKarmakar committed:
attempt for #451

Modified the env.reset() function to add the reward_space and observation_space parameters. Modified the reset() func of the subclasses. Added a OptionalArgumentValue class in gym_type_hints.py for the default value.

ChrisCummins commented:
Hi @SoumyajitKarmakar, sounds great! Would you like me to review your changes or are you still working on this?

Cheers,
Chris

SoumyajitKarmakar commented:
Hi @ChrisCummins,
Yes, please review my changes and let me know if it requires any additional modifications.

ChrisCummins commented on tests/compiler_env_test.py:
```diff
@@ -147,9 +147,9 @@ def test_switch_default_reward_space_in_episode(env: LlvmEnv):
     episode reward.
     """
-    env.reward_space = None
+    # env.reward_space = None
```
Nitpick: Don't commit uncommented code, just remove the line

ChrisCummins commented on tests/compiler_env_test.py:
```diff
@@ -167,9 +167,9 @@ def test_set_same_default_reward_space_in_episode(env: LlvmEnv):
     cumulative episode reward if the reward space is unchanged.
     """
-    env.reward_space = "IrInstructionCount"
+    # env.reward_space = "IrInstructionCount"
```
Same again here

ChrisCummins reviewed (commented):
Hi @SoumyajitKarmakar, thanks for your patch, your changes look good!

Since you are adding new features, could you please add unit tests to cover these new features? I think `tests/compiler_env_test.py` might be a good place to add them, and you can take a look at the existing tests for inspiration.

A couple of other minor nitpicks left inline. Once you push these changes, please ping me for another review.

Cheers,
Chris

SoumyajitKarmakar committed:
Added some unit tests for the new features

Added some simple unit test to test if the modified reset() function is working as intended.

SoumyajitKarmakar commented:
Hi @ChrisCummins,
Please review the latest changes. I have removed the unnecessary commented lines and added some tests.
It is my first time writing tests for a software. I have added 4 simple tests which should hopefully cover the new features. Please let me know if any other kinds of tests are required.

ChrisCummins commented on tests/compiler_env_test.py:
```diff
@@ -34,6 +34,30 @@ def test_benchmark_set_in_reset(env: LlvmEnv):
+    env.reward_space = "IrInstructionCount"
+    assert env.reward_space != "IrInstructionCount"
+    env.reset()
+    assert env.reward_space == "IrInstructionCount"
```
For this and the other 4 assertions you added, you must convert `env.reward_space` and `env.observation_space` to strings to check their value:

```suggestion
    assert str(env.reward_space) == "IrInstructionCount"
```

SoumyajitKarmakar replied:
Oh my bad, fixed them.

ChrisCummins reviewed (commented):
Thanks for adding the tests! The test logic looks good, you just need to make sure to convert the `env.reward_space` and `env.observation_space` attributes to strings for comparison.

Once you fix that, this looks ready to merge.

Cheers,
Chris

SoumyajitKarmakar committed:
Fixed the mistakes

ChrisCummins commented:
Thanks @SoumyajitKarmakar for fixing this :) This LGTM now, but holding off on merging until I fix #657.

Cheers,
Chris

ChrisCummins commented:
Ah, looks like I gave you bad advice on how to fix the failing tests. Not your fault, I'll patch this up myself and merge via #659.

Thanks for contributing to CompilerGym @SoumyajitKarmakar! 🎉 

Cheers,
Chris

SoumyajitKarmakar commented:
It was a pleasure working with you.
